src/analysis/Environments/PauseButton.ts

```diff
@@ -0,0 +1,39 @@
+import { ReactNode } from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { getDisplayStatus, isComputePausable } from 'src/analysis/utils/resource-utils';
+import { getToolLabelFromCloudEnv, isPauseSupported } from 'src/analysis/utils/tool-utils';
+import { Link } from 'src/components/common';
+import { makeMenuIcon } from 'src/components/PopupTrigger';
+import { App } from 'src/libs/ajax/leonardo/models/app-models';
+import { ListRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+
+import { LeoResourcePermissionsProvider } from './Environments.models';
+
+type PausePermissionsProvider = Pick<LeoResourcePermissionsProvider, 'canPauseResource'>;
+
+interface PauseButtonProps {
+  cloudEnvironment: App | ListRuntimeItem;
+  permissions: PausePermissionsProvider;
+  pauseComputeAndRefresh: (cloudEnvironment: App | ListRuntimeItem) => void;
+}
+
+export function PauseButton(props: PauseButtonProps): ReactNode {
+  const { cloudEnvironment, permissions, pauseComputeAndRefresh } = props;
+  const shouldShowPauseButton =
+    isPauseSupported(getToolLabelFromCloudEnv(cloudEnvironment)) && permissions.canPauseResource(cloudEnvironment);
+
+  return shouldShowPauseButton
+    ? h(
+        Link,
+        {
+          style: { marginRight: '1rem' },
+          disabled: !shouldShowPauseButton,
```

Comment:
This should be equal to `isComputePausable`, right? I notice that the Pause button on each row in Runtimes is always enabled even for paused runtimes.

Reply:
a good catch, but the logic is existing and not part of the intended scope of this PR.  A focused PR that just addresses this would be good for the team to dig into.